programs/openbook-twap/src/lib.rs

```diff
@@ -177,6 +178,44 @@ pub struct CancelOrder<'info> {
     pub openbook_program: Program<'info, OpenbookV2>,
 }
 
+#[derive(Accounts)]
+pub struct PruneOrders<'info> {
+    // This is a permissionless function but could be made
+    // to require the close_market_rent_receiver's signature
+    // pub signer: Signer<'info>,
+    pub twap_market: Account<'info, TWAPMarket>,
+    /// CHECK: verified in CPI
+    #[account(mut)]
+    pub open_orders_account: UncheckedAccount<'info>,
+    pub market: AccountLoader<'info, Market>,
+    #[account(mut)]
+    pub bids: AccountLoader<'info, BookSide>,
+    #[account(mut)]
+    pub asks: AccountLoader<'info, BookSide>,
+    pub openbook_program: Program<'info, OpenbookV2>,
+}
+
+#[derive(Accounts)]
+pub struct CloseMarket<'info> {
+    /// CHECK: This is a permissionless function but could be made to require the close_market_rent_receiver's signature
+    #[account(mut)]
+    pub close_market_rent_receiver: UncheckedAccount<'info>,
+    #[account(has_one = close_market_rent_receiver)]
+    pub twap_market: Account<'info, TWAPMarket>,
+    #[account(mut)]
+    pub market: AccountLoader<'info, Market>,
+    #[account(mut)]
+    pub bids: AccountLoader<'info, BookSide>,
+    #[account(mut)]
+    pub asks: AccountLoader<'info, BookSide>,
+    /// CHECK: verified in CPI
+    #[account(mut)]
+    pub event_heap: UncheckedAccount<'info>,
+    /// CHECK: verified in CPI
+    pub token_program: UncheckedAccount<'info>,
```

Comment:
think we may as well verify this since we'll get better error message from client if we screw this up, right?

Reply:
For sure, can add the check. I was just copy pasting what you had in the other structs with minimal modifications.

Reply:
Actually, in all instances of `token_program` in the different account structs you leave the checking of token program up to the openbook program. I think leaving it to openbook is mostly fine. Openbook uses anchor and will clearly state what account was passed in wrong. Plus it's possible they have support for token_2022 somehow, though I'm not sure. So better to leave it to them.